Mathlib/Algebra/Group/Pointwise/Set/Card.lean

```diff
@@ -3,60 +3,79 @@ Copyright (c) 2024 Yaël Dillies. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yaël Dillies
 -/
-import Mathlib.Algebra.Group.Pointwise.Finset.Basic
+import Mathlib.Data.Set.Pointwise.Finite
 import Mathlib.SetTheory.Cardinal.Finite
 
 /-!
-# Cardinalities of pointwise operations on sets.
+# Cardinalities of pointwise operations on sets
 -/
 
-namespace Set
-
-open Pointwise
-
-variable {α β : Type*}
+open scoped Cardinal Pointwise
 
-section MulAction
-variable [Group α] [MulAction α β]
+namespace Set
+variable {G M α : Type*}
 
-@[to_additive (attr := simp)]
-lemma card_smul_set (a : α) (s : Set β) : Nat.card ↥(a • s) = Nat.card s :=
-  Nat.card_image_of_injective (MulAction.injective a) _
+section Mul
+variable [Mul M] {s t : Set M}
 
-end MulAction
+@[to_additive]
+lemma card_mul_le : #(s * t) ≤ #s * #t := by rw [← image2_mul]; exact Cardinal.mk_image2_le
 
-section IsCancelMul
-variable [Mul α] [IsCancelMul α] {s t : Set α}
+variable [IsCancelMul M]
 
 @[to_additive]
-lemma card_mul_le : Nat.card (s * t) ≤ Nat.card s * Nat.card t := by
+lemma natCard_mul_le : Nat.card (s * t) ≤ Nat.card s * Nat.card t := by
```

Comment:
Arguably deprecations are needed here and throughout this PR.

Reply:
The names are reused